### PR TITLE
Feature/swe 553

### PR DESCRIPTION
**Pull request by hughes036**
**Overview:**
This PR converts the FUA MD5 compute from using CryptoJS -> Spark JS.  

Big thanks and credit to @pgarrison for spot on analysis, finding SparkJS, on prem testing, and help debugging. 

**Goal:**
This change produces performance gains > 100%, and achieves FSS2 > FSS.current performance:

image
http://confluence.corp.alleninstitute.org/display/SF/FSS2+Benchmark+Testing

**Testing:**
Unit tests, manual testing (over VPN, and on prem for benchmarking)

**Reviewing:**
`MD5Hasher.ts`

### CODE REVIEW

File: src/renderer/containers/Table/CustomCells/StatusCell/index.tsx [hunk 1]
```diff
@@ -99,7 +99,9 @@ export default function StatusCell(props: CellProps<UploadSummaryTableRow>) {
       step = Step.TWO;
       displayForStep = "0";
       totalForStep = "1";
-    } else if (bytesUploaded && totalBytes) {
+    } 
+    // If uploading bytes, and progress has been made, display progress
```
Comment (hughes036) [inline src/renderer/containers/Table/CustomCells/StatusCell/index.tsx:103]: Unrelated to focus of pr.  An improvement made on wrong branch during an separate pr review.

**Review by kmitcham — APPROVED** [review]

**Review by pgarrison — APPROVED** [review]

File: src/renderer/containers/Table/CustomCells/StatusCell/index.tsx (hunk 1 above)
Comment (SeanLeRoy) [reply src/renderer/containers/Table/CustomCells/StatusCell/index.tsx:103]: The formatting of this is weird, could you move it down below the if?

File: src/renderer/services/file-management-system/Md5Hasher.ts [hunk 2]
```diff
@@ -1,100 +1,46 @@
...
-     * Finalizes the hash computation.
-     * Note that the finalize operation is effectively a destructive, read-once operation.
-     *
-     * @param messageUpdate (Optional) A final message update.
-     *
-     * @return The hash.
-     */
-    finalize(messageUpdate?: CryptoJS.lib.WordArray | string): CryptoJS.lib.WordArray;
-}
+import * as SparkMD5 from "spark-md5";
```
Comment (SeanLeRoy) [inline src/renderer/services/file-management-system/Md5Hasher.ts:1]: I think you only use the `ArrayBuffer` so you could just import that like so:
```
import { ArrayBuffer } from "spark-md5";
```

Comment (hughes036) [reply src/renderer/services/file-management-system/Md5Hasher.ts:1]: Done.

File: src/renderer/containers/Table/CustomCells/StatusCell/index.tsx (hunk 1 above)
Comment (hughes036) [reply src/renderer/containers/Table/CustomCells/StatusCell/index.tsx:103]: pushed a change, lmk if it is good now.

File: src/renderer/services/file-management-system/ChunkedFileReader.ts [hunk 3]
```diff
@@ -74,7 +74,7 @@ export default class ChunkedFileReader {
       highWaterMark: readStreamChunkSize,
     });
 
-    let hasher = new Md5Hasher();
+    let hasher: Md5Hasher = new Md5Hasher();
```
[the remark this answers is not in the document]
Comment (hughes036) [reply src/renderer/services/file-management-system/ChunkedFileReader.ts]: reverted

**Pull request merged**